src/components/Form.js

```diff
@@ -171,14 +171,16 @@ class Form extends React.Component {
      * @returns {Object} - An object containing the errors for each inputID, e.g. {inputID1: error1, inputID2: error2}
      */
     validate(values) {
+        const trimmedValues = {};
+        _.each(values, (inputValue, inputID) => _.isString(inputValue) && (trimmedValues[inputID] = inputValue.trim()));
```

Comment:
won't this crash the app when we're trimming an empty string?

Reply:
We just updated _.isString(inputValue) into !_.isEmpty(inputValue) So it also resolved this problem

Reply:
Oh wait me, I need to check again

Reply:
@rushatgabhane 
I don't see that `emptyString.trim() ---> crash 💥`,  emptyString.trim() = ""
The crash app can be caused from some boolean fields
In the first commit, I added the condition _.isString(inputValue) === true to avoid the crash
And I just updated _.isString(inputValue) into !_.isEmpty(inputValue) So it also resolved this problem because _.isEmpty(boolean) === true

Reply:
@rushatgabhane Bump

Reply:
sorry for the back and forth but this will break when we pass a number

Reply:
can we handle this gracefully please
1. handle non string values
2. trim string values

Reply:
@rushatgabhane 
I agree that

> agree with you. but then there's a reason you're checking if the value is string before trimming.
> 
> Form is a generic component so I think we should handle non-string values as well.

But we only can trim a string value so I suggest that we should come up with the first implementation that adds a condition `_.isString(inputValue)` before trimming and doing nothing if it is a non-string value.

Could you give me an example that we should trim a non-string value?

Reply:
@tienifr here is what i mean 
allValues = [...trimmedStringValues, ...nonTrimmedNonStringValues]

Reply:
we aren't gonna trim non string values.
but we shouldn't drop completely them either

Reply:
@rushatgabhane Oh, my bad 😄 Just updated

Reply:
@rushatgabhane bump